github/Requester.py

```diff
@@ -833,6 +844,51 @@ def __structuredFromJson(self, data: str) -> Any:
                     raise
                 return {"data": data}
 
+    def getFile(
+        self,
+        url: str,
+        path: str,
+        parameters: Optional[Dict[str, Any]] = None,
+        headers: Optional[Dict[str, str]] = None,
+        cnx: Optional[Union[HTTPRequestsConnectionClass, HTTPSRequestsConnectionClass]] = None,
+    ) -> None:
+        """
+        GET a file from the server and save it to the given path.
+        """
+        _, _, stream_chunk_iterator = self.getResponseStream(url, parameters, headers, cnx)
+        with open(path, "wb") as f:
+            for chunk in stream_chunk_iterator():
+                if chunk:
+                    f.write(chunk)
+
+    def getResponseStream(
+        self,
+        url: str,
+        parameters: Optional[Dict[str, Any]] = None,
+        headers: Optional[Dict[str, str]] = None,
+        cnx: Optional[Union[HTTPRequestsConnectionClass, HTTPSRequestsConnectionClass]] = None,
+    ) -> Tuple[int, Dict[str, Any], Callable]:
+        """
+        GET a stream from the server.
+
+        :returns:``(status, headers, stream_chunk_iterator)``
```

Comment:
Here, `stream_chunk_iterator` is technically not true, it returns a function that provides the iterator.

If we'd wire `chunk_size` through (after the `cnx` argument) all the way from `download_asset`, we could get away with returning the iterator directly here. I think this would simplify the API. Plus, `chunk_size` is also meaningful for `getFile`. and `download_asset(path="...", chunk_size=...)`. Then, `stream_chunk_iterator` can stay here.

Reply:
this now works as you suggested